extensions/notebook/src/jupyter/serverInstance.ts

```diff
@@ -377,8 +377,14 @@ export class PerNotebookServerInstance implements IServerInstance {
 		return childProcess;
 	}
 
-	private getEnvWithConfigPaths(env: {}): any {
-		let newEnv: { [key: string]: string } = Object.assign({}, env);
+	private getEnvWithConfigPaths(env: { [key: string]: string }): any {
+		// Merge the current process.env object with the specified object so that we can pass information to notebooks
+		// Variables in the env object will overwrite the ones in process.env object
+		let newEnv: { [key: string]: string } = Object.assign({}, process.env);
+		for (let key in env) {
```

Comment:
Object.assign(newEnv, env); 
the above will simply copy env go newEnv.

Reply:
And even more succinctly you can just add it to the original statement :

```
let newEnv: { [key: string]: string } = Object.assign({}, process.env, env)
```